### DISCUSSION TITLE
EventPattern timing

### DISCUSSION THREAD

**Discussion by macduff111** [supriya-project/supriya#360] [Q&A] [answered]
Hi
I'm playing around with EventPatterns and have worked out that my synthdef needs a gate and that I need to set delta and duration keys in the EventPattern  (which are the equivalent of delta and sustain in sclang). However, the timings of the result are too slow by a factor of ~2. What am I missing? My test code below. I expect 4 notes each a second long in series, but they are each about 2 secs long.

```
@synthdef()
def test(gate=1,freq=440):
	env=Linen.kr(gate,0.01,1.0,0.01,done_action=DoneAction.FREE_SYNTH)
	osc=SinOsc.ar(frequency=freq)
	out=env*osc
	Out.ar(bus=0,source=out)

pattern=EventPattern(
	freq=SequencePattern([440,880],2),
	delta=1,
	duration=1,
	synthdef=test)
```

**Comment by josiah-wolf-oberholtzer** [comment, marked as answer]
See my comment in https://github.com/josiah-wolf-oberholtzer/supriya/issues/362

> Timing's not quite the same as with SuperCollider.

> 1.0 duration is a whole note.

> The default clock (`Clock.default()` will get it for you) defaults to 120BPM. That's 2 beats per second, making 4 beats (4/4) take 2 seconds.
